cirq-core/cirq/transformers/eject_phased_paulis_test.py

```diff
@@ -53,29 +54,42 @@ def assert_optimizes(
             )
 
     # And match the expected circuit.
-    assert circuit == expected, (
-        "Circuit wasn't optimized as expected.\n"
-        "INPUT:\n"
-        "{}\n"
-        "\n"
-        "EXPECTED OUTPUT:\n"
-        "{}\n"
-        "\n"
-        "ACTUAL OUTPUT:\n"
-        "{}\n"
-        "\n"
-        "EXPECTED OUTPUT (detailed):\n"
-        "{!r}\n"
-        "\n"
-        "ACTUAL OUTPUT (detailed):\n"
-        "{!r}"
-    ).format(before, expected, circuit, expected, circuit)
+    cirq.testing.assert_same_circuits(circuit, expected)
 
     # And it should be idempotent.
     circuit = cirq.eject_phased_paulis(
         circuit, eject_parameterized=eject_parameterized, context=context
     )
-    assert circuit == expected
+    cirq.testing.assert_same_circuits(circuit, expected)
```

Comment:
Move here and below to own test ?

Reply:
The nested cases are pretty complex (as we have already seen in the bugfixes in previous PRs), so I prefer this approach where we test the nested case for every non-nested case (eg: with / without ignore tags, with / without eject_parameterized etc.)

Reply:
SG